teleop:
  walk:
    type: topic
    message_type: geometry_msgs/Twist
    topic_name: cmd_vel
    deadman_buttons: [ 4 ]
    axis_mappings:
      - axis: 3
        target: angular.z
        scale: 1.5
      - axis: 1
        target: linear.x
        scale: 0.3
      - axis: 0
        target: linear.y
        scale: 0.3

  body_rotation:
    type: topic
    message_type: geometry_msgs/Twist
    topic_name: body_rotation
    axis_mappings:
      - axis: 3
        target: angular.x
        scale: -0.261
      - axis: 1
        target: angular.y
        scale: 0.523
      - axis: 0
        target: angular.z
        scale: 0.523

  bodyrotation_switch:
    type: topic
    message_type: std_msgs/Float32
    topic_name: bodyrotation_switch
    deadman_buttons: [ 1 ]
    axis_mappings:
      - button: 1
        target: data
        scale: 2
        offset: 0

  load_controller:
    type: topic
    message_type: std_msgs/Float32
    topic_name: load_controller
    deadman_buttons: [ 7 ]
    axis_mappings:
      - button: 7
        target: data
        scale: 2
        offset: 0
  
  emergency_stop:
    type: topic
    message_type: std_msgs/Float32
    topic_name: emergency_stop
    deadman_buttons: [ 3 ]
    axis_mappings:
      - button: 3
        target: data
        scale: 2
        offset: 0

  set_walk:
    type: topic
    message_type: std_msgs/Float32
    topic_name: set_walk
    deadman_buttons: [ 5 ]
    axis_mappings:
      - button: 5
        target: data
        scale: 2
        offset: 0
  
  stance_switch:
    type: topic
    message_type: std_msgs/Float32
    topic_name: stance_switch
    deadman_buttons: [ 5, 0 ]
    axis_mappings:
      - button: 0
        target: data
        scale: 2
        offset: 1